### PR TITLE
Adjust monster immunities to venoms, being downed

### PR DESCRIPTION
**Pull request by Venera3**
#### Summary

SUMMARY: Balance "Adjust monster immunities, buff player venoms"

#### Purpose of change

While cleaning up after my venom rebalance I noticed that the way monster immunities to effects were calculated was neither intuitive nor realistic. This PR attempts to change that, at least with regards to poisons and bleeding.

#### Describe the solution

- Make all zombies impervious to systemic venoms, ie. the old poisons and wasp/bee venom, while keeping them susceptible to direct tissue damage
- Make giant arthropods susceptible to venoms and bleeding in general

- [x] Rework player venoms into something more gameplay-relevant than applying the old venom effects
- [x] Add immunity to being downed to flying monsters as well as critters with the bodytypes Blob, Fish, and Snake and a 75% resist to Multi-legged critters ( Insect, Spider, Crab bodytypes )
- [x] Figure out which of the more exotic critters should get which immunity
- [x] Rewrite the tests 

#### Describe alternatives you've considered

Keep zombies generally poisonable, but for a lot of poisons that simply makes little sense ( venoms that don't directly damage the tissue tend to affect systems the zombies don't have anymore like circulation etc.).
Not messing with the conditions and adding WARM to all insects to at least have them be poisoned, but then the new venoms would have affected everything from a bot to triffids.
Rework the immunity check into something JSONable instead of directly referencing the effect ids, but that's a bit intimidating and would probably count as a "Feature".

#### Testing

The intended mobs get poisoned. Hitting NPCs / getting hit by Apis applies the non-player venom variants. The tests pass.

#### Additional context

In the current implementation, both bleeding and poison require that the affected mob is warm-blooded (?) *and* made of flesh (bleeding) or flesh or iflesh (poisons). Since no bugs get the WARM flag they are effectively immune to both, as are by the way skeletons (who are also not warm, for some reason).

My general assumptions:
- Alive animals should be able to be poisoned and bled ( the player is one, and we get both );
- Giant bugs should be able to be poisoned bled ( most insect venom is developed against other insects, bugs are full of haemolymph and the butchery results reference a more mammal-like physiology anyway );
- <strike>Mycus, Triffids and other weird stuff should not be poisoned and should not bleed ( too different physiology to justify poisons working, they could use some extra effects just for them down the line)</strike>;
- Zombies should not be able to be poisoned by systemic poisons, but tissue damage should work on them;

### CODE REVIEW

**Comment by actual-nh** [comment]
Note prior discussion in #46079.

**Comment by Venera3** [comment]
The fact that this touches on pretty fundamental design questions is the other reason I put it up separately. Since they are energized by transdimensional Blob voodoo, don't need to breathe and evolve rapidly the assumption that their physiology is *pretty* different is a safe one. Still, their material is flesh and they don't seem to want to change that in the course of the first year, even though they incorporate extraneus material, so I'd suppose proteolytic enzymes would keep working for a while, at least.

**Comment by actual-nh** [comment]
Heh. Those design questions (@I-am-Erk? @kevingranade? anyone?) had me considering which Discussions category to ask about them in... thanks!

**Comment by actual-nh** [comment]
BTW, weirdly enough, Magiclysm [lizardfolk](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/mods/Magiclysm/monsters/lizardfolk.json) (and their pet crocodiles!) have NO_BREATHE....

**Comment by Venera3** [comment]
First pass on the player venoms, very WIP:
- Venomous attacks do 10 dmg on application and then 1 dmg with a 50% chance each tick, 1 minute duration from one hit with a 2 min maximal duration
- Very Venomous does 20 dmg on application, 4 dmg with 50% chance, and applies a hefty -50 speed debuff, also one minute per hit up to max 2 mins,

Those feel relevant, is slightly underpowered for monsters ( up from completely irrelevant, so that's progress). However, since the players are working with effectively 80ish max health for each limb, and a third of the limbs are an instakill if they break that would be pretty unfair to say the least. Because of this, the mutated attacks now apply the locally damaging mob venom effect or both the damaging and the weakening venom when hitting an NPC or a player.

**Comment by Venera3** [comment]
Nether creatures and power leeches are now immune to the same venoms zombies got immune to, but those made of flesh can still bleed. Player venoms have a 20% chance to cause Stun / Downed, for the weaker and stronger variant respectively.

Looking at it once more I'm not convinced making zombies immune to some venoms but not to others is good game design even if it makes my inner pedant happier, so I'm opening it for review in the hopes of some core dev feedback.

**Comment by Venera3** [comment]
While I'm here I also did a sanity check on the "downed" effect, mostly applied by Sweeping attacks so that it can't affect flying things or things without y'know, legs to sweep. Arthropods also got a 75% immunity to it. Included that immunity in the tests as well.
This will also mean that I'll need to fiddle a bit more with the strong player venoms' secondary effect.

File: src/monster.cpp [hunk 1]
```diff
@@ -111,7 +117,7 @@ static const trait_id trait_FLOWERS( "FLOWERS" );
 static const trait_id trait_KILLER( "KILLER" );
 static const trait_id trait_MYCUS_FRIEND( "MYCUS_FRIEND" );
 static const trait_id trait_PACIFIST( "PACIFIST" );
-static const trait_id trait_PHEROMONE_INSECT( "PHEROMONE_INSECT" );
+static const trait_id trait_PHEROMONE_( "PHEROMONE_INSECT" );
```
Comment (actual-nh) [inline src/monster.cpp:120]:
```suggestion
static const trait_id trait_PHEROMONE_INSECT( "PHEROMONE_INSECT" );
```

**Comment by Night-Pryanik** [comment]
This will conflict with #46357. Bleeding will be overhauled in that PR, so maybe you cancel the changes to bleeding immunity and leave only your changes to poison immunity?

**Comment by Night-Pryanik** [comment]
#46357 was merged. @Venera3 could you please resolve conflicts?

**Comment by Venera3** [comment]
> #46357 was merged. @Venera3 could you please resolve conflicts?

Yeah, I'm just about done with those. However, I added some test cases for fungaloids and flaming eyes to check for poisons/downing and included bleed in them. Neither has a species with a declared blood type or a relevant flag, but *both* can still bleed ( both in game and in the tests). Did you test them as well?

**Comment by actual-nh** [comment]
Side note: If a wasp-specific mutation line ever gets created, another player venom-type dualistic manifestation: NPCs could logically get pain (bradykinin); not sure about monsters.

File: tests/creature_effect_test.cpp [hunk 2]
```diff
@@ -351,39 +354,128 @@ TEST_CASE( "monster is_immune_effect", "[creature][monster][effect][immune]" )
...
+            CHECK_FALSE( graboid.is_immune_effect( effect_venom_dmg ) );
+            CHECK_FALSE( graboid.is_immune_effect( effect_venom_player1 ) );
+            CHECK_FALSE( graboid.is_immune_effect( effect_venom_player2 ) );
+            CHECK_FALSE( graboid.is_immune_effect( effect_venom_weaken ) );
+        }
+
+        THEN( "they can't be downed" ) {
+            CHECK( graboid.is_immune_effect( effect_downed ) );
+        }
+
```
Comment (Night-Pryanik) [inline tests/creature_effect_test.cpp:396]:
```suggestion
```

**Comment by Night-Pryanik** [comment]
> Neither has a species with a declared blood type or a relevant flag, but both can still bleed ( both in game and in the tests). Did you test them as well?

I didn't test them myself, but tests in your own PR say they _cant't_ bleed.

**Comment by Venera3** [comment]
CHECK: is_imunne returns false, since the can bleed. Nevertheless, here's a screenie from the latest exp build (11326, clean download):
Bleed

The problem has to lie in their materials, since fungaloids bleed plant sap and flaming eyes normal blood from materials veggy and flesh respectively. Your test checked a generator with a steel mat, and riot bots also don't bleed.

**Comment by Night-Pryanik** [comment]
Ok. Fungaloids _do_ bleed as they have `veggy` material, though @kevingranade [stated](https://github.com/CleverRaven/Cataclysm-DDA/pull/45000#issuecomment-720273025) that they shouldn't bleed at all. We should fix this sometime, but I think it's out of scope of this PR.
Flaming eyes _do_ bleed too as they have `flesh` material.

So I'd say it's working as intended (except for fungi bleeding), and tests should pass on these monsters.

**Comment by actual-nh** [comment]
@Venera3: You may find [this PMC article](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6891279/) of interest regarding uses of venom other than predation/feeding and defense, BTW.

**Pull request merged**